### PR TITLE
Instructions on Porch Tracing setup added to Porch contributor guide

### PR DESCRIPTION
**Pull request by liamfallon**

### CODE REVIEW

File: content/en/docs/porch/contributors-guide/environment-setup.md [hunk 1]
```diff
@@ -179,3 +179,74 @@ and running the [setup script](https://github.com/nephio-project/porch/blob/main
...
+
+2. Add the environment variable `OTEL` to the porch-server manifest:
+
+```
+kubectl edit deployment -n porch-system porch-server
+```
+
+```
+env:
+   - name: OTEL
```
Comment (JamesMcDermott) [inline content/en/docs/porch/contributors-guide/environment-setup.md:202]: Could de-indent the `name` line a few spaces to align it with the `value` line.
(I suppose we can expect the deployment will have other `env` variables as an example, meaning this comment will be just for aesthetics in this file)

**Review by JamesMcDermott — COMMENTED** [review]
Just a few nitpicky little things...

File: content/en/docs/porch/contributors-guide/environment-setup.md [hunk 2]
```diff
@@ -179,3 +179,74 @@ and running the [setup script](https://github.com/nephio-project/porch/blob/main
...
+## Enabling Open Telementry/Jaeger tracing
+
+### Enabling tracing on a Porch deployment
+
+Follow the steps below to enable Open Telemetry/Jaeger tracing on your Porch deployment.
+
+1. Apply the Porch deployment.yaml manifest for Jaeger.
+
+```
+kubectl apply -f kubectl apply -f https://raw.githubusercontent.com/nephio-project/porch/refs/heads/main/deployments/tracing/deployment.yaml
```
[the remark this answers is not in the document]
Comment (liamfallon) [reply content/en/docs/porch/contributors-guide/environment-setup.md]: done.

File: content/en/docs/porch/contributors-guide/environment-setup.md (hunk 1 above)
Comment (liamfallon) [reply content/en/docs/porch/contributors-guide/environment-setup.md]: Done.

File: content/en/docs/porch/contributors-guide/environment-setup.md [hunk 3]
```diff
@@ -179,3 +179,74 @@ and running the [setup script](https://github.com/nephio-project/porch/blob/main
...
+2. Run Jaeger:
+
+```
+cd jaeger
+./jaeger-all-in-one
+```
+
+3. Configure the Porch server to output Open Telemetry traces:
+
+Set the `OTEL` environment variable to point at the Jaeger server
```
Comment (CsatariGergely) [inline content/en/docs/porch/contributors-guide/environment-setup.md:229]:
```suggestion
Set the *OTEL* environment variable to point at the Jaeger server
```

File: content/en/docs/porch/contributors-guide/environment-setup.md [hunk 4]
```diff
@@ -179,3 +179,74 @@ and running the [setup script](https://github.com/nephio-project/porch/blob/main
...
+
+In `.vscode/launch.json`:
```
Comment (CsatariGergely) [inline content/en/docs/porch/contributors-guide/environment-setup.md:231]:
```suggestion
In *.vscode/launch.json*:
```

File: content/en/docs/porch/contributors-guide/environment-setup.md [hunk 5]
```diff
@@ -179,3 +179,74 @@ and running the [setup script](https://github.com/nephio-project/porch/blob/main
...
+
+```
```
Comment (CsatariGergely) [inline content/en/docs/porch/contributors-guide/environment-setup.md:233]:
```suggestion
```bash
```

File: content/en/docs/porch/contributors-guide/environment-setup.md [hunk 6]
```diff
@@ -179,3 +179,74 @@ and running the [setup script](https://github.com/nephio-project/porch/blob/main
...
+   ...
+"OTEL": "otel://localhost:4317",
+   ...
+   ...
+}
+```
+
+In a shell:
+
+```
```
Comment (CsatariGergely) [inline content/en/docs/porch/contributors-guide/environment-setup.md:245]:
```suggestion
```bash
```

File: content/en/docs/porch/contributors-guide/environment-setup.md [hunk 7]
```diff
@@ -179,3 +179,74 @@ and running the [setup script](https://github.com/nephio-project/porch/blob/main
...
+export OTEL="otel://localhost:4317"
+```
+
+4. Open the Jaeger UI in your browser at http://localhost:16686
```
Comment (CsatariGergely) [inline content/en/docs/porch/contributors-guide/environment-setup.md:249]:
```suggestion
4. Open the Jaeger UI in your browser at *http://localhost:16686*
```

**Review by CsatariGergely — CHANGES_REQUESTED** [review]

File: content/en/docs/porch/contributors-guide/environment-setup.md [hunk 8]
```diff
@@ -179,3 +179,74 @@ and running the [setup script](https://github.com/nephio-project/porch/blob/main
...
+2. Run Jaeger:
+
+```bash
+cd jaeger
+./jaeger-all-in-one
+```
+
+3. Configure the Porch server to output Open Telemetry traces:
+
+Set the *OTEL* environment variable to point at the Jaeger server
```
Comment (CsatariGergely) [inline content/en/docs/porch/contributors-guide/environment-setup.md:229]:
```suggestion
   Set the *OTEL* environment variable to point at the Jaeger server
```

File: content/en/docs/porch/contributors-guide/environment-setup.md [hunk 9]
```diff
@@ -179,3 +179,74 @@ and running the [setup script](https://github.com/nephio-project/porch/blob/main
...
+
+In *.vscode/launch.json*:
```
Comment (CsatariGergely) [inline content/en/docs/porch/contributors-guide/environment-setup.md:231]:
```suggestion
   In *.vscode/launch.json*:
```

File: content/en/docs/porch/contributors-guide/environment-setup.md [hunk 10]
```diff
@@ -179,3 +179,74 @@ and running the [setup script](https://github.com/nephio-project/porch/blob/main
...
+"env": {
+   ...
+   ...
+"OTEL": "otel://localhost:4317",
+   ...
+   ...
+}
+```
+
+In a shell:
```
Comment (CsatariGergely) [inline content/en/docs/porch/contributors-guide/environment-setup.md:243]:
```suggestion
   In a shell:
```

**Review by CsatariGergely — CHANGES_REQUESTED** [review]
In this way hopefully the text is rendered in aliment with the numbered list of the steps.

**Pull request merged**